### PR TITLE
fix(DB): Draenei Start Area Class Training Quests Fixes

### LINKED ISSUE #9922: Draenei start zone class quests are available too early
Original Cromiecraft report https://github.com/chromiecraft/chromiecraft/issues/2685 While this wasn't specifically the issue in question it was noticed during triage. After digging deeper I found this.

### What client do you play on?

enUS

### Faction

- [X] Alliance
- [ ] Horde

### Content Phase:

- [ ] Generic
- [X] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59
- [ ] 60

### Current Behaviour

Class trainers can immediately give the quest "[Insert Class] Training" with no prerequisite in the Dreanei starting zone

### Expected Blizzlike Behaviour

These quests should not be available until after completing the quest ["Replenishing the Healing Crystals" (Quest 9280)](https://tbc.wowhead.com/quest=9280/replenishing-the-healing-crystals) This quest is a part of a chain for all of the class quests ([Mage (quest 9290)](https://tbc.wowhead.com/quest=9290/mage-training), [Hunter (quest 9288)](https://tbc.wowhead.com/quest=9288/hunter-training), [Paladin (quest 9287)](https://tbc.wowhead.com/quest=9421/shaman-training), [Priest (quest 9291)](https://tbc.wowhead.com/quest=9291/priest-training), [Shaman (Quest 9421)](https://tbc.wowhead.com/quest=9421/shaman-training) and [Warrior (Quest 9289)](https://tbc.wowhead.com/quest=9289/warrior-training#comments:id=280700) in the draenei start zone

### Source

https://tbc.wowhead.com/quest=9280/replenishing-the-healing-crystals
https://wowpedia.fandom.com/wiki/Paladin_Training

### Steps to reproduce the problem

1. Roll a new draenei paladin/priest/mage/hunter/shaman/warrior 
2. Walk down the path (or alternately use gm commands) and immediately go talk to your corresponding trainer: Paladin=Aurelon (.go c 57212), Priest=Zalduun (.go c 57213), Mage=Valaatu (.go c 84581), Hunter= Keilnei (.go c 84585), Shaman=Firmanvarr (.go c 59513) Warrior= Kore (.go c 84574)
3. Click the quest "[Class] Training"

### Extra Notes

Original issue was for the Paladin quest not having text upon acceptance.  Additionally all of the class quests above do not have acceptance text, only completion text.
https://youtu.be/WdvYXH6Jc3Q 

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/33673bc1c2864cf9dd86209bd915d4529ec83f00

### Operating system

Windows 10

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chat-transmitter](https://github.com/azerothcore/mod-chat-transmitter)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-custom-worldboss](https://github.com/55Honey/Acore_CustomWorldboss)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)
- [lua-zone-debuff](https://github.com/55Honey/Acore_ZoneDebuff)

### Customizations

None

### Server

ChromieCraft

### PR DESCRIPTION
**Pull request by ZhengPeiRu21**
## Changes Proposed:
Fix three different issues with these quest:
-  Quest should require Replenishing the Healing Crystals completed
-  Quest should be can auto-complete when talk to the trainer and see quest complete text; no need for accept then complete - this is why no LogDescription or QuestDescription can be find for these quests
- Quests should all only be available to Draenei. Before was inconsistent.

## Issues Addressed:

- Closes #9922 

## SOURCE:

https://tbc.wowhead.com/quest=9290/mage-training
https://tbc.wowhead.com/quest=9288/hunter-training
https://tbc.wowhead.com/quest=9421/shaman-training
https://tbc.wowhead.com/quest=9291/priest-training
https://tbc.wowhead.com/quest=9421/shaman-training
https://tbc.wowhead.com/quest=9289/warrior-training

## Tests Performed:

- Created character, quest is not present. Complete quest 9280, quest is present with question mark icon. Completion text is correctly show. Test with non-Draenei character, quest is not available.

## How to Test the Changes:

1. Create Draenei character, level to 2
2. Visit trainer, see quest is not available
3. Complete quest 9280, quest is now available
4. Quest text is good completion text, no blank Accept text box.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Review by acidmanifesto — APPROVED** [review]

**Comment by acidmanifesto** [comment]
Same as the other pr, i guess the sql check changed at some point. please move the notes above the individual querey or remove them for the code style check can pass so i can mark it for merge.

File: data/sql/updates/pending_db_world/rev_1643757425749275400.sql [hunk 1]
```diff
@@ -0,0 +1,4 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1643757425749275400');
+UPDATE `quest_template_addon` SET `PrevQuestID` = 9280 WHERE `id` IN (9288, 9290, 9287, 9291, 9421, 9290); /* Training quests should require complete Replenishing the Healing Crystals */
```
Comment (acidmanifesto) [inline data/sql/updates/pending_db_world/rev_1643757425749275400.sql:2]: move the notes above or delete them. i guess the code style changed at some point.

**Review by acidmanifesto — CHANGES_REQUESTED** [review]

File: data/sql/updates/pending_db_world/rev_1643757425749275400.sql (hunk 1 above)
Comment (ZhengPeiRu21) [reply data/sql/updates/pending_db_world/rev_1643757425749275400.sql]: I have removed the comments. Thanks!

**Review by acidmanifesto — APPROVED** [review]

**Comment by acidmanifesto** [comment]
@ZhengPeiRu21 do you have a discord account and are you present in the azerothcore discord? I made mention of your phenominal pr contributions to shin and he would like to acknowledge you with some in channel rank.

File: data/sql/updates/pending_db_world/rev_1643757425749275400.sql [hunk 2]
```diff
@@ -0,0 +1,4 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1643757425749275400');
+UPDATE `quest_template_addon` SET `PrevQuestID` = 9280 WHERE `id` IN (9288, 9290, 9287, 9291, 9421, 9290);
+UPDATE `quest_template` SET `Flags` = 65664 WHERE `id` IN (9288, 9290, 9287, 9291, 9421, 9290);
+UPDATE `quest_template` SET `AllowableRaces` = 1024 WHERE `id` IN (9288, 9290, 9287, 9291, 9421, 9290);
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1643757425749275400.sql:4]:
```suggestion
UPDATE `quest_template` SET `Flags` = 65664, `AllowableRaces` = 1024 WHERE `id` IN (9288, 9290, 9287, 9291, 9421, 9290);
```

**Comment by ZhengPeiRu21** [comment]
> @ZhengPeiRu21 do you have a discord account and are you present in the azerothcore discord? I made mention of your phenominal pr contributions to shin and he would like to acknowledge you with some in channel rank.

Thank you! That is very kind of you to mention. My Discord name is 郑佩茹#3132. I am not in AzerothCore Discord but would be glad to join! Please send me an invitation.

File: data/sql/updates/pending_db_world/rev_1643757425749275400.sql (hunk 2 above)
Comment (ZhengPeiRu21) [reply data/sql/updates/pending_db_world/rev_1643757425749275400.sql]: That is much cleaner, thank you for the suggestion!

**Comment by acidmanifesto** [comment]
> > @ZhengPeiRu21 do you have a discord account and are you present in the azerothcore discord? I made mention of your phenominal pr contributions to shin and he would like to acknowledge you with some in channel rank.
> 
> Thank you! That is very kind of you to mention. My Discord name is 郑佩茹#3132. I am not in AzerothCore Discord but would be glad to join! Please send me an invitation.

https://discord.com/invite/bXyHx5gD

**Pull request merged**